Modix.Data/Repositories/MessageRepository.cs

```diff
@@ -86,6 +89,49 @@ public async Task<int> GetGuildUserMessageCount(ulong guildId, ulong userId, Tim
             return result.ToDictionary(x => userQuery[x.Key], x => x.Count());
         }
 
+        public async Task<GuildUserParticipationStatistics> GetGuildUserParticipationStatistics(ulong guildId, ulong userId)
+        {
+            var stats = await ModixContext.Database.GetDbConnection()
```

Comment:
Noooooow I see why you wanted me to look at this.

So. I'm not totally AGAINST using Dapper. But I think there needs to be a really strong argument for pulling in a SECOND Data Access library. And we need buy-in from all the primary maintainers.

I'll have a closer look at the query itself and see if I agree with you that (presumably) it's not doable in EF.

Reply:
I couldn't find a great way to express this with LINQ, and I didn't have a ton of luck figuring out how to run raw SQL (that wasn't associated with a `DbSet`) in EF Core, so I reverted to Dapper. Is there an easy way to do the same without Dapper?

Reply:
After playing with it for a while, I was able to find some success with EF Core with the following changes. Not sure how idiomatic this is, but it :WorksOnMyMachine:
`MessageRepository.cs`: https://paste.mod.gg/ecobinavox.cs
`ModixContext.cs`: https://paste.mod.gg/iqasawemew.cs

Reply:
Updated